### PR TITLE
Routing, Page data & front-end menus

### PR DESCRIPTION
**Pull request by jakxnz**
* https://trello.com/c/WK1LFinJ

### CODE REVIEW

**Comment by jakxnz** [comment]
[Routing, Page data & front-end menus](https://trello.com/c/WK1LFinJ/38-routing-page-data-front-end-menus)

File: package.json [hunk 1]
```diff
@@ -29,10 +29,12 @@
     "@apollo/react-testing": "^4.0.0",
     "@babel/core": "^7.17.8",
     "@babel/eslint-parser": "^7.17.0",
+    "@babel/plugin-transform-modules-commonjs": "^7.17.7",
```
Comment (Krank3n) [inline package.json:32]: wow lots of new packages

File: src/js/App.js [hunk 2]
```diff
@@ -20,9 +20,15 @@ const t = {
...
+const layouts = {
+  home: Home,
+  resorts: Resorts,
+  resort: Resort,
+};
+
 const App = ({ suspenseQuery }) => {
   const [showSidebar, setShowSidebar] = useState(false);
-  const viewData = useMemo(() => ({ showSidebar, setShowSidebar }), [showSidebar]);
+  const viewData = useMemo(() => ({ showSidebar, setShowSidebar, layouts }), [showSidebar]);
```
Comment (Krank3n) [inline src/js/App.js:31]: ooo using the new linting rules.... nice

File: package.json (hunk 1 above)
Comment (jakxnz) [reply package.json:32]: True, I should probably double-check if we ended up using them all. I had to try so many things when troubleshooting. I should have done it on its own branch.

Comment (jakxnz) [reply package.json]: I've cleaned up the `package.json` now 👍

File: src/js/components/Header/Header.js [hunk 3]
```diff
@@ -1,20 +1,12 @@
...
-  CNavItem,
+  CContainer, CHeader, CHeaderToggler, CImage,
 } from '@coreui/react';
 import React, { useContext } from 'react';
-import { FormattedMessage } from 'react-intl';
 import { Link } from 'react-router-dom';
 import Logo from '../../../images/logo.svg';
-import routingConfig from '../config/routing-config';
-import SecondaryHeader from '../SecondaryHeader/SecondaryHeader';
+import HeaderMenuMain from '../HeaderMenuMain/HeaderMenuMain';
```
Comment (Krank3n) [inline src/js/components/Header/Header.js:9]: Why did you add 'Main' to the end of HeaderMenu? Are you looking to add a HeaderMenuSub or something?

File: src/js/components/HeaderMenuMain/HeaderMenuMain.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const HeaderMenuMain = () => {
+  const { items } = useStaticMenu(menuCode);
+
+  if (!items.length) {
+    return null;
+  }
+
+  return (
+    <CHeaderNav className="header-nav d-none d-md-flex me-auto">
+      {items.map(({ title, url, isCurrent }) => (
```
Comment (Krank3n) [inline src/js/components/HeaderMenuMain/HeaderMenuMain.js:18]: Are we going to be using Icons for the Header Menu?

File: src/js/components/DynamicLayout/DynamicLayout.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  const { cmsPage: { layout, meta_title, meta_description } } = data;
+
+  const Component = layouts[layout]; // React will warn us if this isn't a valid component
+
+  return (
+    <>
+      <Helmet>
+        <meta charSet="utf-8" />
+        <title>{ meta_title }</title>
```
Comment (Krank3n) [inline src/js/components/DynamicLayout/DynamicLayout.js:46]: Is this the browser tab title?

like this: 
image

File: src/js/components/DynamicSwitch/DynamicSwitch.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import React, { useEffect } from 'react';
+import { Route, Switch, useLocation } from 'react-router-dom';
+import DynamicLayout from '../DynamicLayout/DynamicLayout';
+
+/**
+ * Restore the scroll everytime we change routes
+ *
+ * A handy component from https://v5.reactrouter.com/web/guides/scroll-restoration
+ */
+const ScrollToTop = () => {
```
Comment (Krank3n) [inline src/js/components/DynamicSwitch/DynamicSwitch.js:12]: Should this be in it's own hook?

I'm gonna guess its gonna be resued?

**Review by Krank3n — COMMENTED** [review]
Hey man good work on this.

I am trying to be more thorough with my PR's after gaining more experience, and maybe because my PR is on the line :D

I noticed a couple of things:

Is this chome browser title {is it a meta tag?} for the resort supposed to change dynamically?
I noticed it changed for the resorts (list) page.
image

On the resorts (list) page... the cards are really wide, however if open up the home inspection and adjust the size of the inspection window, the resort card goes to the normal size.
image

image
Are you able to replicate this bug?

File: src/js/components/Header/Header.js (hunk 3 above)
Comment (jakxnz) [reply src/js/components/Header/Header.js:9]: Yep, it just keeps the name space explicit enough to add other menus without renaming out-of-scope files. e.g `HeaderMenuSecondary`

File: src/js/components/HeaderMenuMain/HeaderMenuMain.js (hunk 4 above)
Comment (jakxnz) [reply src/js/components/HeaderMenuMain/HeaderMenuMain.js:18]: I um'd and ah'd about this, and decided that this would best be left out of scope, because:

- There's no designs, that I'm aware of
- The icon backend Trello task hasn't been picked up yet
- Therefor, there's no way to determine what or how icons are applied within the scope of routing

File: src/js/components/DynamicLayout/DynamicLayout.js (hunk 5 above)
Comment (jakxnz) [reply src/js/components/DynamicLayout/DynamicLayout.js:46]: Yep, it's this guy https://www.w3schools.com/tags/tag_title.asp

File: src/js/components/DynamicSwitch/DynamicSwitch.js (hunk 6 above)
Comment (jakxnz) [reply src/js/components/DynamicSwitch/DynamicSwitch.js:12]: I wondered this, and decided to apply a "first use then re-use" approach.

I can't think of another scenario that would use this. I'm aware that the resorts list paginator scrolls back to a top, but I can't say for sure what scroll top options that needs, therefor I figured this was light enough to leave as is for use to refactor down the line if it's causing any overhead.

**Comment by jakxnz** [comment]
Loving the level of detail man! Great questions all 'round

> Is this chome browser title {is it a meta tag?} for the resort supposed to change dynamically? I noticed it changed for the resorts (list) page. image

I designed a few approaches to this, but ended up deciding the solution for this would make this story too big, so I raised a Trello card for it: https://trello.com/c/Wq86poVO

> On the resorts (list) page... the cards are really wide, however if open up the home inspection and adjust the size of the inspection window, the resort card goes to the normal size. image
> 
> image Are you able to replicate this bug?

Yeah good question. I can take a look to see if the scope of these changes has effected this. If this changeset hasn't caused a regression, I'll raise it as a bug. If it has, I'll investigate further.

**Comment by jakxnz** [comment]
> I can take a look to see if the scope of these changes has affected this.

I've investigated, and at first glance, this seems like its behaviour as the `.scss` instructs it to. I sent you a Slack message https://shredindex.slack.com/archives/DSUFBFUEA/p1649803777344639

File: src/stories/components/Header/Header.stories.js [hunk 7]
```diff
@@ -11,14 +14,47 @@ export default {
...
+        variables: {
+          code: headerMenuCode,
+        },
+      },
+      result: {
+        data: {
+          staticMenu: {
+            menuItems: [
+              {
+                title: 'Foo',
```
Comment (Krank3n) [inline src/stories/components/Header/Header.stories.js:30]: Are we able to use more realistic test names? :)

Comment (Krank3n) [reply src/stories/components/Header/Header.stories.js:30]: Im just being unlogiclaly fussy

**Review by Krank3n — APPROVED** [review]
